### PR TITLE
Cosmos Module simplification

### PR DESCRIPTION
**Pull request by NicolasMahe**
Summary of this PR:
- remove cosmos app and cosmos app factory structure 🍾 
- simplify cosmos module helper
- separate initialization of App and BasicManager
- BasicManager using the modules' AppModuleBasic struct is stateless and enough to generate the genesis or any client (CLI). No need to init the whole app the generate the genesis or start the cosmos CLI
- init cosmos app (using modules of engine and cosmos) in `sdk/modules.go`
- separate sdk's backend to keeper (logic on the data) and module (module's handler and querier).

### CODE REVIEW

File: sdk/modules.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		codec.Codec,
+		paramsStoreKey,
+		paramsKeeper.Subspace(auth.DefaultParamspace),
+		auth.ProtoBaseAccount,
+	)
+	bankKeeper := bank.NewBaseKeeper(
+		accountKeeper,
+		paramsKeeper.Subspace(bank.DefaultParamspace),
+		bank.DefaultCodespace,
+		nil,
```
Comment (NicolasMahe) [inline sdk/modules.go]: Add `ModuleAccountAddrs`
See https://github.com/cosmos/scaffold/blob/master/templates/lvl-1/app/app.go.tmpl#L141

File: sdk/execution/module.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		case "get":
+			hash, err := hash.Decode(path[1])
+			if err != nil {
+				return nil, err
+			}
+			return k.Get(request, hash)
+		case "list":
+			return k.List(request)
+		default:
+			return nil, errors.New("unknown execution query endpoint" + path[0])
```
Comment (krhubert) [inline sdk/execution/module.go]:
```suggestion
			return nil, fmt.Errorf("unknown execution query endpoint %s", path[0])
```

File: sdk/instance/module.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			}
+			return k.List(request, &f)
+		case "exists":
+			hash, err := hash.Decode(path[1])
+			if err != nil {
+				return nil, err
+			}
+			return k.Exists(request, hash)
+		default:
+			return nil, errors.New("unknown instance query endpoint" + path[0])
```
Comment (krhubert) [inline sdk/instance/module.go]:
```suggestion
			return nil, fmt.Errorf("unknown instance query endpoint %s", path[0])
```

File: sdk/ownership/module.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	}
+}
+
+func querier(k *Keeper) cosmos.Querier {
+	return func(request cosmostypes.Request, path []string, req abci.RequestQuery) (res interface{}, err error) {
+		switch path[0] {
+		case "list":
+			return k.List(request)
+		default:
+			return nil, errors.New("unknown ownership query endpoint" + path[0])
```
Comment (krhubert) [inline sdk/ownership/module.go]:
```suggestion
			return nil, fmt.Errorf("unknown ownership query endpoint %s", path[0])
```

File: sdk/process/module.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		case "get":
+			hash, err := hash.Decode(path[1])
+			if err != nil {
+				return nil, err
+			}
+			return k.Get(request, hash)
+		case "list":
+			return k.List(request)
+		default:
+			return nil, errors.New("unknown service query endpoint" + path[0])
```
Comment (krhubert) [inline sdk/process/module.go]:
```suggestion
			return nil, fmt.Errorf("unknown service query endpoint %s", path[0])
```

File: sdk/runner/module.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		case "get":
+			hash, err := hash.Decode(path[1])
+			if err != nil {
+				return nil, err
+			}
+			return k.Get(request, hash)
+		case "list":
+			return k.List(request)
+		default:
+			return nil, errors.New("unknown runner query endpoint" + path[0])
```
Comment (krhubert) [inline sdk/runner/module.go]:
```suggestion
			return nil, fmt.Errorf("unknown runner query endpoint %s", path[0])
```

File: sdk/service/module.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+			return k.Hash(&createServiceRequest), nil
+		case "exists":
+			hash, err := hash.Decode(path[1])
+			if err != nil {
+				return nil, err
+			}
+			return k.Exists(request, hash)
+
+		default:
+			return nil, errors.New("unknown service query endpoint" + path[0])
```
Comment (krhubert) [inline sdk/service/module.go]:
```suggestion
			return nil, fmt.Errorf("unknown service query endpoint %s", path[0])
```

**Pull request merged**